shared/python/trial_util.py

```diff
@@ -75,12 +75,17 @@ def run_trials(method, task_name,
                 writer.writeheader()
 
             for args in product(*parameter_ranges):
+                narg = list(args)
+                if narg and narg[0] == 'cpu':
+                    pid = os.getpid()
+                    print(f'Pinning PID: {pid}')
+                    os.system(f'taskset -p 0x000000ff {pid}')
```

Comment:
Separate question: Is there a reason this has to be done inside `run_trials` and not before calling the `run.sh` of the experiment? I think it would be better to separate this from the actual running of trials

Reply:
Thanks for the suggestions. Sounds good to make separate configuration files.

For the place of the code, one problem is that to pin the process, we need to know the pid of the python process that actually runs the experiments (there can be multiple python processes at a time). So if we place the code right above where the trial starts, we can guarantee that the process is pinned for experiments using CPU. One thing we can try is to pin the process in `run_relay.py` and other similar files before the actual experiments start.

Reply:
If you do the pinning in a parent process, is the child process also pinned? If so, you can have a Bash script called (let's say) `run_pin.sh` that pins itself (a Bash process can get its own pid with `$$`) and calls `run.sh`

Reply:
This is a good question. I'll try to see whether this works.

Reply:
Separate note on specifying which processors to pin (since it came up in our discussion): The `man` page for `taskset` describes how to specify the CPUs to use. Besides the bitmask, there is also the `--cpu-list` option that takes CPU indices (perhaps we should configure that too instead of hardcoding it?)

Reply:
Much simpler way to handle processor pinning for in `run.sh`: The `man` page for `taskset` also says that there is another interface for the command that allows for _launching_ a command with the processors specified (`taskset [options] mask command [argument...]`)

Reply:
That's a great point. We can try running the command with taskset then.